src/bin/fish.rs

```diff
@@ -71,9 +71,99 @@
 
 const DOC_DIR: &str = env!("DOCDIR");
 const DATA_DIR: &str = env!("DATADIR");
+const DATA_DIR_SUBDIR: &str = env!("DATADIR_SUBDIR");
 const SYSCONF_DIR: &str = env!("SYSCONFDIR");
 const BIN_DIR: &str = env!("BINDIR");
 
+#[cfg(feature = "installable")]
+fn install(noconfirm: bool) {
+    use rust_embed::RustEmbed;
+
+    #[derive(RustEmbed)]
+    #[folder = "share/"]
+    struct Asset;
+
+    use std::fs;
+    use std::io::ErrorKind;
+    use std::io::Write;
+    use std::io::{stderr, stdin};
+    let (_, home) = fish::env::get_user_home();
+    let Some(home) = home else {
+        eprintln!("Can't find $HOME",);
+        std::process::exit(1);
+    };
+    let dir = PathBuf::from(home).join(DATA_DIR).join(DATA_DIR_SUBDIR);
+
+    // TODO: Translation,
+    // FLOG?
+    // - Install: Translations
+    // - Install: Manpages (build via build.rs)
+    // - Don't install: __fish_build_paths.fish.in
+    if !noconfirm {
+        eprintln!(
+            "This will write fish's data files to '{}'.\n\
+             Please enter 'yes' to continue.",
+            dir.display()
+        );
+        eprint!("> ");
+        let _ = stderr().flush();
+
+        let mut input = String::new();
+        stdin()
+            .read_line(&mut input)
+            .expect("Failed to read from stdin");
```

Comment:
should not crash here but print the error and `process::exit(1)`.
On some terminals it's easy to enter invalid UTF-8 by accident etc.
(We changed our panic handler to create coredumps so preferably only internal errors should crash)